### PR TITLE
Secondary graph flow

### PR DESCRIPTION
**Pull request by ajrgrubbs**
Supports drawing graphs where one is the child of another.

If you've reviewed this project before, make sure to `rm /tmp/vee_reports/*` to ensure your data sample is structured correctly.

### CODE REVIEW

**Comment by adamyala** [comment]
👍

File: vee/tree.html [hunk 1]
```diff
@@ -14,9 +14,19 @@
...
-      if (err) { throw err; }
+    var categoryClass = function(d) {
+      return d.data.category || 'process';
+    };
+
+    var genUniqueIdFn = function(rootData) {
+      return function(d) {
+        var localUnique = d.data.process_id ? "proc" + d.data.process_id : d.data.id;
+        var globalUnique = `${rootData.process_id}_${localUnique}`;
+        return globalUnique;
```
Comment (adamyala) [inline vee/tree.html]: Why not `return ${rootData.process_id}_${localUnique};`? Do we use `globalUnique` elsewhere?

I already tossed in my thumbs up. Just curious if storing this then returning it is a legit performance hit.

Comment (ajrgrubbs) [inline vee/tree.html]: Nah, global unique isn't used elsewhere. I think I wrote it that way simply to make it easier to step through with chrome's debugger.
It shouldn't have any performance hit though. From what I know about this kind of stuff, either way it's going to be creating a reference to where this string is in memory and returning that reference to the function caller. Maybe javascript needs to do an extra lookup in the variable map, but that should be constant-time.

File: vee/tree.html [hunk 2]
```diff
@@ -14,9 +14,19 @@
...
-    d3.json('/report/393', function(err, data) {
-      if (err) { throw err; }
+    var categoryClass = function(d) {
+      return d.data.category || 'process';
+    };
+
+    var genUniqueIdFn = function(rootData) {
+      return function(d) {
+        var localUnique = d.data.process_id ? "proc" + d.data.process_id : d.data.id;
+        var globalUnique = `${rootData.process_id}_${localUnique}`;
```
Comment (Prajaktcs) [inline vee/tree.html]: Will the template strings `` cause any issue with older browsers that don't support ECMA6?

**Comment by csavage1994** [comment]
👍

File: vee/tree.html (hunk 2 above)
Comment (ajrgrubbs) [inline vee/tree.html]: Yes, but I've been using a lot more of ES6 than the template strings. Constants and Classes make up a good portion of the codebase. If I need to go to an older compatibility that will need to be it's own PR.

Comment (Prajaktcs) [inline vee/tree.html]: 👍
